sklearn/ensemble/weight_boosting.py

```diff
@@ -522,8 +521,8 @@ def _boost_real(self, iboost, X, y, sample_weight, random_state):
 
         # Boost weight using multi-class AdaBoost SAMME.R alg
         estimator_weight = (-1. * self.learning_rate
-                                * (((n_classes - 1.) / n_classes) *
-                                   inner1d(y_coding, np.log(y_predict_proba))))
+                            * ((n_classes - 1.) / n_classes)
+                            * (y_coding * np.log(y_predict_proba)).sum(axis=1))
```

Comment:
I found that `np.einsum('ij,ij->i', y_coding, np.log(y_predict_proba))` was another possibility but I find it harder to grok (maybe because I don't use einsum very regularly).

Note that (at least in the tests) both `y_coding` and `y_predict_proba` shapes are `(n, p)`.

Reply:
Can't this be done with `np.dot`?

Reply:
I don't see how but maybe I am missing a numpy trick. In the tests `y_coding` and `y_predict_proba` are 2d of shape `(n_samples, n_classes)` and you want an output that is of shape `(n_samples,)`.

Reply:
Right